drivers/segger/common_rtt.c

```diff
@@ -0,0 +1,37 @@
+/****************************************************************************
+ * drivers/segger/common_rtt.c
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/spinlock.h>
+
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
+
+struct rspinlock_s g_segger_lock = RSPINLOCK_INITIALIZER;
+
+/****************************************************************************
```

Comment:
There are no public or private functions, but one of those sections must exist in order to pass nxstyle. That's why the empty "Public Functions" is added in here.

Reply:
we found the similar problem, could you wait this patch get merged?
https://github.com/apache/nuttx/pull/16512/commits/1547cc4a9765dc3a3876605078de66b9806acffb
and then apply the recursive spinlock change to avoid the conflict.